### PR TITLE
Add collector for the systemd_units table

### LINKED ISSUE #30790: Add collector for the systemd_units table
| Wazuh version | Install type | Install method     | Platform |
|---------------|--------------|--------------------|----------|
| 4.14.0        | Agent        | Packages / Sources | Linux    |

# Description

This issue proposes the implementation of a new Linux-only collector for the `systemd_units` inventory table within the `syscollector` module. This collector will use the `extended_sources` submodule to retrieve metadata about all loaded `systemd` units, providing insight into system service definitions, runtime states, and their associated configuration paths.

The information collected will be mapped to the inventory model and formatted in accordance with ECS-compatible schemas where applicable. This will help improve visibility into system configuration, service availability, and audit trails for service state changes.

**Relevant fields:**

| Column           | Type    | Description                                                                 |
|------------------|---------|-----------------------------------------------------------------------------|
| `id`             | text    | Unique unit identifier                                                      |
| `description`    | text    | Unit description                                                            |
| `load_state`     | text    | Whether the unit definition was properly loaded                             |
| `active_state`   | text    | High-level unit activation state (e.g., active, inactive)                   |
| `sub_state`      | text    | Low-level unit activation state (depends on unit type)                      |
| `unit_file_state`| text    | Unit enablement state (e.g., enabled, disabled, masked)                     |
| `following`      | text    | Name of another unit that this unit follows                                 |
| `object_path`    | text    | DBus object path for the unit                                               |
| `job_id`         | bigint  | Next queued job ID                                                          |
| `job_type`       | text    | Type of the job (e.g., start, stop)                                         |
| `job_path`       | text    | DBus object path for the job                                                |
| `fragment_path`  | text    | Path to the unit file this unit was read from                               |
| `user`           | text    | Configured user for the unit, if any                                        |
| `source_path`    | text    | Path to the generated or original unit configuration                        |

# Tasks

- [x] Implement a new `SystemdUnitsProvider` in `extended_sources`  
- [x] Normalize collected data to match the target inventory schema  
- [x] Add unit and integration tests to validate the collector on supported Linux distributions  

# Acceptance Criteria

- Systemd unit metadata is successfully retrieved and structured in JSON format  
- Collector works on all systemd-based Linux distributions (Ubuntu, RHEL, Fedora, etc.)  
- Tests confirm correct behavior and coverage of edge cases  

# References

- [osquery systemd_units table](https://github.com/osquery/osquery/blob/master/specs/linux/systemd_units.table)  
- [Table group reference](https://docs.google.com/spreadsheets/d/1XwgDInDeReqhZ6DoAn4SmqrKvCFmN40NjoUBWQiwt54/edit?gid=0#gid=0)  
- [Wazuh Agent Inventory Enrichment Plan](https://github.com/wazuh/wazuh-agent/issues/664#issuecomment-2769043629)

### PR DESCRIPTION
**Pull request by nbertoldo**
## Description

Closes #30790 

Implements a new Linux-only collector for the `systemd_units` inventory table within the `syscollector` module. This collector will use the `extended_sources` submodule to retrieve metadata about all loaded `systemd` units, providing insight into system service definitions, runtime states, and their associated configuration paths.

The information collected will be mapped to the inventory model and formatted in accordance with ECS-compatible schemas where applicable. This will help improve visibility into system configuration, service availability, and audit trails for service state changes.

**Relevant fields:**

| Column           | Type    | Description                                                                 |
|------------------|---------|-----------------------------------------------------------------------------|
| `id`             | text    | Unique unit identifier                                                      |
| `description`    | text    | Unit description                                                            |
| `load_state`     | text    | Whether the unit definition was properly loaded                             |
| `active_state`   | text    | High-level unit activation state (e.g., active, inactive)                   |
| `sub_state`      | text    | Low-level unit activation state (depends on unit type)                      |
| `unit_file_state`| text    | Unit enablement state (e.g., enabled, disabled, masked)                     |
| `following`      | text    | Name of another unit that this unit follows                                 |
| `object_path`    | text    | DBus object path for the unit                                               |
| `job_id`         | bigint  | Next queued job ID                                                          |
| `job_type`       | text    | Type of the job (e.g., start, stop)                                         |
| `job_path`       | text    | DBus object path for the job                                                |
| `fragment_path`  | text    | Path to the unit file this unit was read from                               |
| `user`           | text    | Configured user for the unit, if any                                        |
| `source_path`    | text    | Path to the generated or original unit configuration                        |

## Proposed Changes

- Implement a new `SystemdUnitsProvider` class in `extended_sources` module.
- Add unit tests to validate the new class.
- Add the required wrappers to allow unit testing.

### Results and Evidence

<details><summary>Source item example</summary>

```json
{
    "active_state": "inactive",
    "description": "Pollinate to seed the pseudo random number generator",
    "following": "",
    "fragment_path": "/usr/lib/systemd/system/pollinate.service",
    "id": "pollinate.service",
    "job_id": "0",
    "job_path": "/",
    "job_type": "",
    "load_state": "loaded",
    "object_path": "/org/freedesktop/systemd1/unit/pollinate_2eservice",
    "source_path": "",
    "sub_state": "dead",
    "unit_file_state": "enabled",
    "user": "pollinate"
}
```
</details> 

<details><summary>Systemd_units collector item example</summary>

```json
{
    "active_state": "inactive",
    "description": "Pollinate to seed the pseudo random number generator",
    "following": "",
    "fragment_path": "/usr/lib/systemd/system/pollinate.service",
    "id": "pollinate.service",
    "job_id": 0,
    "job_path": "/",
    "job_type": "",
    "load_state": "loaded",
    "object_path": "/org/freedesktop/systemd1/unit/pollinate_2eservice",
    "source_path": "",
    "sub_state": "dead",
    "unit_file_state": "enabled",
    "user": "pollinate"
}
```
</details>

- [source_systemd_units.log](https://github.com/user-attachments/files/21323000/osquery_systemd_units.log)
- [wazuh_systemd_units.log](https://github.com/user-attachments/files/21323003/wazuh_systemd_units.log)

## Review Checklist

- [x] Code changes reviewed
- [x] Relevant evidence provided
- [x] Tests cover the new functionality
- [ ] Configuration changes documented
- [ ] Developer documentation reflects the changes
- [x] Meets requirements and/or definition of done
- [x] No unresolved dependencies with other issues

### CODE REVIEW

File: src/data_provider/src/extended_sources/services/CMakeLists.txt [hunk 1]
```diff
@@ -29,14 +32,13 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
...
+    set(PLATFORM_LIBS_SERVICES dbus-1 pthread)
   endif()
 endif()
 
 add_library(services STATIC ${SRC_FILES})
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
-  target_link_libraries(services ${PLATFORM_LIBS_SERVICES})
-endif()
+target_link_libraries(services ${PLATFORM_LIBS_SERVICES})
```
Comment (cborla) [inline src/data_provider/src/extended_sources/services/CMakeLists.txt:41]: Shouldn't we filter it so that it only comes in through Windows and Linux? To avoid other operating systems?

File: src/data_provider/src/extended_sources/services/src/systemd_units_linux.cpp [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    if (!m_dbusWrapper->message_iter_init(reply, &iter))
+    {
+        std::cerr << "Reply has no arguments." << std::endl;
+        m_dbusWrapper->message_unref(reply);
+        return false;
+    }
+
+    if (DBUS_TYPE_ARRAY != m_dbusWrapper->message_iter_get_arg_type(&iter))
+    {
+        std::cerr << "Expected an array." << std::endl;
```
Comment (cborla) [inline src/data_provider/src/extended_sources/services/src/systemd_units_linux.cpp]: For the time being, we should not include error messages.

File: src/data_provider/src/extended_sources/services/CMakeLists.txt (hunk 1 above)
Comment (nbertoldo) [reply src/data_provider/src/extended_sources/services/CMakeLists.txt:41]: No need, since I have added the filter here: https://github.com/wazuh/wazuh/blob/7658174772311ca4dff0ba830f6d1adb8d6fab61/src/data_provider/src/extended_sources/CMakeLists.txt#L4

File: src/data_provider/src/extended_sources/services/src/systemd_units_linux.cpp [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+                }
+            }
+        }
+
+        output.emplace_back(std::move(unit));
+        m_dbusWrapper->message_iter_next(&arrayIter);
+    }
+
+    m_dbusWrapper->message_unref(reply);
+    return true;
```
Comment (cborla) [inline src/data_provider/src/extended_sources/services/src/systemd_units_linux.cpp:187]: To avoid possible leaks, it would also free up memory before exiting via true.
`m_dbusWrapper->error_free(&err);`

**Review by cborla — APPROVED** [review]
LGTM

**Comment by cborla** [comment]
The check [Syscollector test on macOS / build (pull_request)](https://github.com/wazuh/wazuh/actions/runs/16891305312/job/47851490553?pr=30996) It will fail until the fix is uploaded to branch 4.14.0.
Fix:
- https://github.com/wazuh/wazuh/pull/31269

Regarding package generation checks, we depend on the update of the project's external dependencies. For now, we can merge it and continue:
- https://github.com/wazuh/wazuh/actions/runs/16936952367/job/48018904371?pr=31000
- https://github.com/wazuh/wazuh/actions/runs/16936952367/job/48018904383?pr=31000
